lib/OpenQA/Worker/Cache/Service.pm

```diff
@@ -0,0 +1,135 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::Cache::Service;
+
+use Mojolicious::Lite;
+use OpenQA::Worker::Cache::Task::Asset;
+use Mojolicious::Plugin::Minion;
+use OpenQA::Worker::Cache;
+use Mojo::Collection;
+use Mojolicious::Plugin::Minion::Admin;
+
+BEGIN { srand(time) }
+
+my $tk;
+my $enqueued = Mojo::Collection->new;
+app->hook(
+    before_server_start => sub {
+        $tk       = int(rand(999999999999));
+        $enqueued = Mojo::Collection->new;
+    });
+
+
+sub SESSION_TOKEN { $tk }
+
+plugin 'Minion' => {SQLite => 'sqlite:' . OpenQA::Worker::Cache->from_worker->db_file};
+plugin 'Minion::Admin';
+plugin 'OpenQA::Worker::Cache::Task::Asset';
+
+sub _gen_guard_name { join('.', SESSION_TOKEN, pop) }
+sub _exists { !!(defined $_[0] && exists $_[0]->{total} && $_[0]->{total} > 0) }
+
+sub active { !app->minion->lock(shift, 0) }
+sub get_job_by_token {
+    eval {
+        { app->minion->backend->list_jobs(0, 1, {note => {token => shift}})->{jobs}[0] }
+    }
+}
+
+sub enqueued {
+    my $lock = shift;
+    !!($enqueued->grep(sub { $_ eq $lock })->size == 1);
+}
+
+sub dequeue {
+    my $lock = shift;
+    $enqueued = $enqueued->grep(sub { $_ ne $lock });
+}
+
+sub enqueue {
+    push @$enqueued, shift;
+}
+
+sub run {
+    shift;
+    $ENV{'MOJO_LISTEN'} ||= 'http://127.0.0.1:7844/';
```

Comment:
Not sure if i want OpenQA::Utils involved here at all - really i don't see the purpose of using that function, and nor importing OpenQA::Utils there -  if not at least giving as extra checking if actually the port was used or not.

Reply:
ok

Reply:
Well, we introduced that function to abstract the IPv6 vs IPv4 differences.

Reply:
I don't see the actual need to bind on ipv6 here - but sure if you wish i can change that

Reply:
I'm just saying there are benefits to use it. I don't think 127.0.0.1 will go away - and as long as you control all the clients (aka workers) to also hardcode 127.0.0.1, it will be fine. But if you mix localhost and 127.0.0.1 it will become problematic.

Reply:
But to add what benefit? The client needs to generate anyway the ip where to connect to. And i don't see ipv6 playing a role here in the future.

The code is explictly trying to set those two values in obvious points to avoid confusion. 

- Service level: https://github.com/os-autoinst/openQA/pull/1783/files/05795388ed020c54a19ca50bffe7eff0b31f3752#diff-4d72ec8e16ff560c6db4bc1ccdaaa0c6R68
- Client level: https://github.com/os-autoinst/openQA/pull/1783/files/05795388ed020c54a19ca50bffe7eff0b31f3752#diff-405c69e4e6cd9a2c943949cea32a451bR24

But sure, for better consistency makes a point.